app/action.py

```diff
@@ -0,0 +1,14 @@
+def action(knights_dict: dict) -> dict:
```

Comment:
This looks non readable

Reply:
Try to use separate functions that calculates your logic here

Reply:
I dont want to use classes here. Changed the logic of calculating the results immediately in pairs